Objective-C/CBLDatabase.mm

```diff
@@ -178,12 +178,16 @@ - (NSString*) description {
 
 
 - (NSString*) path {
-    return _c4db != nullptr ? sliceResult2FilesystemPath(c4db_getPath(_c4db)) : nil;
+    @synchronized (self) {
```

Comment:
I can think of one use case that developers also do synchronized (database). I think it might be better not to do with self. I will make changes.